test/gen-template/matter3/miscellaneous_helper_tests.zapt

```diff
@@ -30,6 +30,14 @@ TargetStruct item {{index}} from Access Control cluster: {{name}}
 TargetStruct item {{index}} from Binding cluster: {{name}}
 {{/zcl_struct_items_by_struct_and_cluster_name}}
 
+{{#zcl_struct_items_by_struct_and_cluster_name "SemanticTagStruct" "Mode Select"}}
+SemanticTagStruct item {{index}} from Mode Select cluster: {{name}}
+{{/zcl_struct_items_by_struct_and_cluster_name}}
+
+{{#zcl_struct_items_by_struct_and_cluster_name "SemanticTagStruct" ""}}
```

Comment:
This test should be passing a nonempty string that is not "Mode Select" (e.g. "Descriptor" to really match what Matter SDK does) to test the desired behavior of this helper.